File: server/schemas/summary.py
from pydantic import BaseModel, Field


class ContractSummaryParty(BaseModel):
    # None when the model named a party the KG has no node for — worth seeing,
    # since it means the extractor missed an entity.
    partyId: str | None = None
    name: str
    role: str = ""
    does: str = ""


class ContractSummary(BaseModel):
    documentId: str
    documentName: str = ""
    title: str = ""
    contractType: str = ""
    # Party mentions are marked `{{partyId|short text}}` so the client colours
    # them without having to match names back to the graph.
    summary: str = ""
    parties: list[ContractSummaryParty] = Field(default_factory=list)

File: web/src/types/summary.ts
export interface ContractSummaryParty {
	/** Null when the abstract names an entity the knowledge graph has no node for. */
	partyId: string | null;
	name: string;
	role: string;
	does: string;
}

export interface ContractSummary {
	documentId: string;
	documentName: string;
	title: string;
	contractType: string;
	/** Party mentions arrive as `{{partyId|short text}}`; see MENTION in ContractAbstract. */
	summary: string;
	parties: ContractSummaryParty[];
}
